app/uk/gov/hmrc/customs/declarations/stub/controllers/UpscanStubController.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.declarations.stub.controllers
+
+import play.api.http.{ContentTypes, HeaderNames}
+import play.api.mvc.{Action, Codec, MessagesControllerComponents}
+import play.api.Logging
+import uk.gov.hmrc.customs.declarations.stub.config.AppConfig
+import uk.gov.hmrc.customs.declarations.stub.models.upscan._
+import uk.gov.hmrc.customs.declarations.stub.models.upscan.Field._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import javax.inject.{Inject, Singleton}
+import scala.xml._
+
+@Singleton
+class UpscanStubController @Inject()(appConfig: AppConfig, httpClient: HttpClient, mcc: MessagesControllerComponents) extends BackendController(mcc) with Logging {
+
+  implicit val ec = mcc.executionContext
+
+  def waiting(ref: String) = {
+    Waiting(
+      UploadRequest(
+        href = s"${appConfig.upscanBaseUrl}/upscan/s3-bucket",
+        fields = Map(
+          Algorithm.toString -> "AWS4-HMAC-SHA256",
+          Signature.toString -> "xxxx",
+          Key.toString -> "xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
+          ACL.toString -> "private",
+          Credentials.toString -> "ASIAxxxxxxxxx/20180202/eu-west-2/s3/aws4_request",
+          Policy.toString -> "xxxxxxxx==",
+          SuccessRedirect.toString -> s"${appConfig.cdsFileUploadFrontendBaseUrl}/cds-file-upload-service/upload/upscan-success/${ref}",
+          ErrorRedirect.toString -> s"${appConfig.cdsFileUploadFrontendBaseUrl}/cds-file-upload-service/upload/upscan-error/${ref}"
+        )
+      )
+    )
+  }
+
+  // for now, we will just return some random
+  def handleBatchFileUploadRequest: Action[NodeSeq] = Action(parse.xml) { implicit req =>
+    val xmlBodyString = req.body.mkString
+    logger.info(s"Batch file upload request: $xmlBodyString")
+    Thread.sleep(100)
```

Comment:
Why do you need to sleep?

Reply:
Honestly I don't know. This code is simply a copy of the existing endpoints already on the SFUS frontend test only routes file. I didn't want to play around too much due to time constraints.